events/models.py

```diff
@@ -431,6 +443,25 @@ def publish(self, send_notifications=True):
     def is_published(self):
         return bool(self.published_at)
 
+    def get_enrolment_count(self) -> int:
+        return sum(
+            occurrence.get_enrolment_count() for occurrence in self.occurrences.all()
+        )
+
+    def get_attended_enrolment_count(self) -> int:
+        return sum(
+            occurrence.get_attended_enrolment_count()
+            for occurrence in self.occurrences.all()
+        )
+
+    def get_capacity(self) -> Optional[int]:
+        try:
+            return sum(
+                occurrence.get_capacity() for occurrence in self.occurrences.all()
+            )
+        except TypeError:
+            return None
```

Comment:
Related to this and all other `get_capacity` functions with the `except TypeError` clause:

Could there be some part in the sum that is None and others that are integers? If that's possible, should we then return the sum of the integers or simply None as here?

Example: [None, 1, 5, None] -> sum should be 1+5=6 or None?

Or is the data model such that the above should not happen? Or is the data somehow wrong if there are Nones and integers both present?

Reply:
I thought about that too. I think that in real world, it should not even be possible that some occurrences does not have a capacity when some others inside the same event have. But, if that would happen, for example because of some test data or something, I think it's better to tell on _event level_, that "we don't know the capacity" than just give something too optimistic.

Reply:
I think this is "external enrolments" vs "internal enrolments" situation. The occurrences should always inherit a default capacity from the event level.